packages/commerce-sdk-react/src/hooks/ShopperProducts/query.ts

```diff
@@ -11,6 +11,24 @@ import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
 
 type Client = ApiClients['shopperProducts']
 
+const productKeys = {
+    all: [{entity: ['product']}],
+    useProducts: (arg: Record<string, unknown>) => [{...productKeys.all[0], scope: 'list', ...arg}],
+    useProduct: (arg: Record<string, unknown>) => [
+        {...productKeys.all[0], scope: 'detail', ...arg},
+    ],
+}
+
+const categoryKeys = {
```

Comment:
Perhaps we can make a factory function that allows us to easily generate `productKeys` and `categoryKeys`. And calling it would be like this:

```js
const categoryKeys = createQueryKeys('category', {useCategories: 'list', useCategory: 'detail'})
const productKeys = createQueryKeys('product', {useProducts: 'list', useProduct: 'detail'})
```

Reply:
I think this is the idea of the new NPM package [@lukemorales/query-key-factory](https://www.npmjs.com/package/@lukemorales/query-key-factory) recently linked from the React Query docs: https://tanstack.com/query/v4/docs/guides/query-keys

I've been testing the package, but so far I see the package is not compatible with the idea mentioned in the previous comment of defining all keys as an array with exactly one object.